src/map/pet.c

```diff
@@ -311,23 +311,19 @@ int pet_performance(struct map_session_data *sd, struct pet_data *pd)
 
 int pet_return_egg(struct map_session_data *sd, struct pet_data *pd)
 {
-	struct item tmp_item;
-	int flag;
+	int i;
 
 	nullpo_retr(1, sd);
 	nullpo_retr(1, pd);
 	pet->lootitem_drop(pd,sd);
-	memset(&tmp_item,0,sizeof(tmp_item));
-	tmp_item.nameid = pd->petDB->EggID;
-	tmp_item.identify = 1;
-	tmp_item.card[0] = CARD0_PET;
-	tmp_item.card[1] = GetWord(pd->pet.pet_id,0);
-	tmp_item.card[2] = GetWord(pd->pet.pet_id,1);
-	tmp_item.card[3] = pd->pet.rename_flag;
-	if((flag = pc->additem(sd,&tmp_item,1,LOG_TYPE_EGG))) {
-		clif->additem(sd,0,0,flag);
-		map->addflooritem(&sd->bl, &tmp_item, 1, sd->bl.m, sd->bl.x, sd->bl.y, 0, 0, 0, 0, false);
+
+	// Pet Evolution
+	for (i = 0; i < MAX_INVENTORY; i++) {
+		if(sd->status.inventory[i].card[0] == CARD0_PET &&
+			pd->pet.pet_id == MakeDWord(sd->status.inventory[i].card[1], sd->status.inventory[i].card[2]))
```

Comment:
probably better add {} because without hard to read

Reply:
Done, converted to ARR_FIND.